### DISCUSSION TITLE
Где пишем отчество пациента

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#31] [Административная РГ] [answered]
(дискуссия перенесена из чата)

**Александр Иванов 25.10.21**

Отчетсво обычно пишут в given, можем ли использовать given? https://www.hl7.org/fhir/datatypes.html#humanname 

Given names (not always 'first'). Includes middle names
This repeating element order: Given Names appear in the correct order for presenting the name
```json
"name": [
    {
        "use": "official",
        "family": "Новосельцева Юрьевна",
        "given": ["Ирина"]
    }
],
```

**Uestas 25.10.21**
6 Отчество переводится как Patronymic, поэтому оно было включено в элемент Family Name (смотри прикрепленный скриншот).

**Александр Иванов 25.10.21**
6. Это хороший аргумент, но я бы предложил посмотреть еще на две вещи:

* Тип данных - given - массив и позволяет нам разделить имя и отчество, а в family нам нужно писать фамилию и отчество через пробел и мы теряем это явное разделение
* Сложившаяся практика - я знаю, что в Хеликс и в проектах Health Samurai независимо принято решение кодировать отчество в given. Предложение кодировать ФИО так в т.ч. зафиксировано в предыдущем предложении по кодированию СЭМД от сообщества - https://fhir-ru.github.io/core/profiles/Patient/remd.html

**Иван Осмоловский 26.10.21**
Добрый день, Александр. По поводу вопроса №6. Мы в свое время сталкивались с похожей проблемой в рамках разработки на CDA. По аналогии мы использовали given для обозначения, как имени, так и отчества. Однако отличить коллегам одно от другого не удавалось. Как Вы решаете данную проблему?

**Тимур Алейников 12 ноября 2021**
По вопросу имен - предложенная конструкция  выглядит очень неудобной для использования. Из нее например неудобно клеить традиционное для РФ: Фамилия Имя Отчество, да и индекс по фамилии нужно делать хитрее, чем просто по полю. Лучше бы отделить отчество. 
```json
"name": [
    {
        "use": "official",
        "family": "Новосельцева Юрьевна",
        "given": ["Ирина"]
    }
],
```

**Comment by niquola** [comment, marked as answer]
В нашей предыдущей версии мы предлагали отчество вторым элементом в given - в Чувашии норм работает
Клеить строчки плохо
```yaml
name:
- given: [Николай, Николаевич]
  family: Рыжиков
```

@alexander-ivanov указал на предыдущее обсуждение - https://github.com/fhir-ru/core/issues/3

**Comment by shamilnn** [comment]
Детально обсуждали здесь: https://github.com/fhir-ru/core/issues/3
В самом типе данных HumanName есть пример с кириллическим ФИО: https://www.hl7.org/fhir/datatypes-examples.html#humanname

**Reply by Ben0063** [reply]
Действительно, сам FHIR предлагает хранить отчество в given[2]. 
Но как быть , когда в такую БД попадет (при каком-нибудь импорте), такой пациент (из того же примера): 
  use value="official"
  family value="Chalmers"
  given value="Peter"
  given value="James"

**Comment by maximalexeev** [comment]
Поддерживаю вариант с двумя элементами given, пользуемся им.

**Comment by niquola** [comment]
Вариант с отчеством как Name.given[1] имеет несколько плюсов

* не надо расширений
* обращаться за отчеством можно `name[1]`
* технически легко собрать полное имя `"{name.family} {name.given.join(' ')}`

**Reply by Ben0063** [reply]
еще нужно уточнить про поиск - по отдельности по имени и по отчеству

**Comment by toropok** [comment]
+1 за то чтобы отчество было в given[1]. работает так в неск проектах и норм

**Comment by Ben0063** [comment]
Мы пока тестируем хранение отчества в стандартном расширении "http://hl7.org/fhir/StructureDefinition/humanname-fathers-family".
Громоздко, но в целом - работать можно.

**Reply by maximalexeev** [reply]
Это расширение не про отчество, по большому счету. Оно про часть **фамилии**, унаследованной у отца. 
Поддерживаю коллегу в оценке, решение плохое.

**Comment by toropok** [comment]
> Мы пока тестируем хранение отчества в стандартном расширении "http://hl7.org/fhir/StructureDefinition/humanname-fathers-family".
> 
> Громоздко, но в целом - работать можно.

на мой взгляд это очень плохо. Поиск отдельно по отчеству вроде бы вообще лишён всякого смысла.
